x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/appearance_selector/appearance_modal.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { type FC } from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  useGeneratedHtmlId,
+  EuiButtonEmpty,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import type { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import type { DarkModeValue as ColorMode } from '@kbn/user-profile-components';
+import { type Value, ValuesGroup } from './values_group';
+import { useAppearance } from './use_appearance_hook';
+
+const systemLabel = i18n.translate('xpack.cloudLinks.userMenuLinks.appearanceModalSystemLabel', {
+  defaultMessage: 'System',
+});
+
+const colorModeOptions: Array<Value<ColorMode>> = [
+  {
+    id: 'system',
+    label: systemLabel,
+    icon: 'desktop',
+  },
+  {
+    id: 'light',
+    label: i18n.translate('xpack.cloudLinks.userMenuLinks.appearanceModalLightLabel', {
+      defaultMessage: 'Light',
+    }),
+    icon: 'sun',
+  },
+  {
+    id: 'dark',
+    label: i18n.translate('xpack.cloudLinks.userMenuLinks.appearanceModalDarkLabel', {
+      defaultMessage: 'Dark',
+    }),
+    icon: 'moon',
+  },
+  {
+    id: 'space_default',
+    label: i18n.translate('xpack.cloudLinks.userMenuLinks.appearanceModalSpaceDefaultLabel', {
+      defaultMessage: 'Space default',
+    }),
+    icon: 'spaces',
+    betaBadgeLabel: i18n.translate('xpack.cloudLinks.userMenuLinks.appearanceModalBetaBadgeLabel', {
+      defaultMessage: 'Deprecated',
+    }),
+    betaBadgeTooltipContent: i18n.translate(
+      'xpack.cloudLinks.userMenuLinks.appearanceModalBetaBadgeTooltip',
+      {
+        defaultMessage: 'Space default settings will be deprecated in 10.0.',
+      }
+    ),
+    betaBadgeIconType: 'warning',
+  },
+];
+
+interface Props {
+  closeModal: () => void;
+  uiSettingsClient: IUiSettingsClient;
+  isServerless: boolean;
+}
+
+export const AppearanceModal: FC<Props> = ({ closeModal, uiSettingsClient, isServerless }) => {
+  const modalTitleId = useGeneratedHtmlId();
+
+  const { onChange, colorMode, isLoading } = useAppearance({
+    uiSettingsClient,
+  });
+
+  return (
+    <EuiModal
+      aria-labelledby={modalTitleId}
+      onClose={closeModal}
+      style={
+        isServerless
+          ? undefined
+          : // When not in serverless, we have the "Space default" as an option.
+            // which renders a warning callout. We don't want the modal to scale up when
+            // the callout is rendered, so we set a fixed width.
+            { width: 600 }
```

Comment:
just curious, is there an EUI provided value for this width that we can access in JS? Perhaps a CSS variable?

Reply:
I followed the EUI docs for the modal width https://eui.elastic.co/#/layout/modal#widths. I don't think there are value for large width like this one.